lib/features/thread/presentation/thread_controller.dart

```diff
@@ -3,57 +3,162 @@ import 'package:core/presentation/utils/responsive_utils.dart';
 import 'package:dartz/dartz.dart';
 import 'package:flutter/material.dart';
 import 'package:get/get.dart';
+import 'package:jmap_dart_client/jmap/account_id.dart';
+import 'package:jmap_dart_client/jmap/core/filter/filter.dart';
+import 'package:jmap_dart_client/jmap/core/sort/comparator.dart';
+import 'package:jmap_dart_client/jmap/mail/email/email.dart';
+import 'package:jmap_dart_client/jmap/mail/email/email_filter_condition.dart';
 import 'package:model/model.dart';
 import 'package:tmail_ui_user/features/base/base_controller.dart';
 import 'package:tmail_ui_user/features/mailbox_dashboard/presentation/mailbox_dashboard_controller.dart';
+import 'package:tmail_ui_user/features/thread/domain/constants/thread_constants.dart';
+import 'package:tmail_ui_user/features/thread/domain/state/get_all_email_state.dart';
+import 'package:tmail_ui_user/features/thread/domain/usecases/get_all_email_interactor.dart';
+import 'package:tmail_ui_user/features/thread/presentation/model/load_more_state.dart';
 import 'package:tmail_ui_user/main/routes/app_routes.dart';
 
 class ThreadController extends BaseController {
 
+  final GetAllEmailInteractor _getAllEmailInteractor;
   final ResponsiveUtils responsiveUtils;
-  final mailboxDashBoardController = Get.find<MailboxDashBoardController>();
 
-  final presentationMailCurrent = PresentationThread.createThreadEmpty().obs;
+  final mailboxCurrent = PresentationMailbox.presentationMailboxEmpty.obs;
+  final selectedEmail = PresentationEmail.presentationEmailEmpty.obs;
 
-  ThreadController(this.responsiveUtils);
+  final emailList = <PresentationEmail>[].obs;
+  final loadMoreState = LoadMoreState.IDLE.obs;
+
+  late Set<Comparator> sortCurrent = Set();
+  late Filter filterCurrent;
+  late AccountId accountId;
+  int positionCurrent = ThreadConstants.POSITION_DEFAULT;
+  int limitCurrent = ThreadConstants.LIMIT_DEFAULT;
```

Comment:
Quite hard to understand all these things for only one use-case: Get emails in one mailbox. Please try to find the better way. Please consider the way for easy debugging and maintaining.

Reply:
Why we need `limitCurrent`?. No more assignment for this variable

Reply:
Same for `sortCurrent`.

Reply:
why we need `accountId` here. Why not place it in dashboard?